[3.2] docs(grid,treelist): documentation for SearchBox enhancements (#877)
telerik/blazor-docs: components/grid/filter/searchbox.md

---
title: Toolbar SearchBox
page_title: Grid - Filtering SearchBox
description: Enable and configure filtering SearchBox in Grid for Blazor.
slug: grid-searchbox
tags: telerik,blazor,grid,filtering,filter,searchbox
published: True
position: 20
---

# Grid Toolbar SearchBox

In addition to the [main filtering options]({%slug components/grid/filtering%}), you can add a SearchBox in the Grid Toolbar.

>caption In this article:

* [Basics](#basics)
* [Filter From Code](#filter-from-code)
* [Customize the SearchBox](#customize-the-searchbox)

## Basics

The SearchBox lets the user type their query and the grid will look up all visible string columns with a case-insensitive `Contains` operator, and filter them accordingly. You can change the filter delay, and the fields the grid will use - see the [Customize the SearchBox](#customize-the-searchbox) section below.

> The SearchBox is independent from the standard filters. If you have filters applied, the SearchBox will amend and respect them. Thus, you can also apply filtering to results returned from it.

Reviewer: amend = change, which may sound misleading

```suggestion
The SearchBox is independent from the standard filters. If you have filters applied, the SearchBox will respect them and add additional filtering criteria. Thus, you can also apply filtering to results returned from it.
```

The `GridSearchBox` component offers the following settings to customize its behavior:

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Attribute | Type and Default Value | Description |
|----------|----------|----------|
| `Class` | `string`| a CSS class rendered on the wrapper of the searchbox so you can customize its appearance.

> | `DebounceDelay` | `int` <br/> (300) | the time in `ms` with which the typing is debounced. This provides a performance optimization when using the `OnRead` event - filtering does not happen on every keystroke anymore.

Reviewer: ```suggestion
| `DebounceDelay` | `int` <br/> (300) | the time in milliseconds with which the typing is debounced. This provides a performance optimization when using the `OnRead` event - filtering does not happen on every keystroke anymore.
```

> | `Fields` |  `List<string>` | A collection of the fields names that the Grid should search in. By default, the Grid searches in all string fields, which are bound to visible columns. You can only define a subset of those fields. It is also possible to programmatically [search in string fields, which are not displayed in the Grid]({%slug grid-kb-search-in-hidden-fields%}).

Reviewer: ```suggestion
| `Fields` |  `List<string>` | The collection of fields to search in. By default, the Grid searches in all string fields, which are bound to visible columns. You can only define a subset of those fields. It is also possible to programmatically [search in string fields, which are not displayed in the Grid]({%slug grid-kb-search-in-hidden-fields%}).
```
